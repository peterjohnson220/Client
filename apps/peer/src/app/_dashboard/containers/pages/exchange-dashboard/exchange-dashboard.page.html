<pf-page-with-title-bar [returnUrl]="'/exchanges/list'">

  <ng-container page-title>
    {{ (exchange$ | async)?.ExchangeName }}
  </ng-container>

  <ng-container page-actions>
    <button type="button" class="btn btn-primary" (click)="mapClick()">Map</button>
    <button type="button" class="btn btn-secondary" (click)="referCompanyClick()">Invite Company</button>
    <button type="button" class="btn btn-secondary" (click)="manageJobsClick()">Manage Jobs</button>
  </ng-container>

  <ng-container page-content>
    <div class="exchange-dashboard-container d-flex flex-row">
      <div class="exchange-dashboard-content d-flex flex-column flex-grow-1">
        <div class="exchange-dashboard-chart-container d-flex flex-row justify-content-around">
          <div class="d-flex flex-grow-1 flex-column justify-content-between mr-2">
            <div class="d-flex card card-body flex-column flex-grow-1 text-center mb-1 justify-content-center">
              <pf-exchange-company-count></pf-exchange-company-count>
            </div>
            <div class="d-flex card card-body flex-column flex-grow-1 text-center mt-1 justify-content-center">
              <pf-exchange-job-count></pf-exchange-job-count>
            </div>
          </div>
          <div class="d-flex flex-grow-2 flex-column justify-content-between mr-2">
            <div class="card card-body flex-column justify-content-top flex-grow-1 text-center">
              <h4>Top Industries</h4>
              <div class="d-flex flex-grow-1 flex-column justify-content-center text-center">
                <pf-exchange-industry-chart></pf-exchange-industry-chart>
              </div>
            </div>
          </div>
          <div class="d-flex flex-grow-2 flex-column justify-content-between mr-2">
            <div class="card card-body flex-column justify-content-top flex-grow-1 text-center">
              <h4>Job Exchange Job Families</h4>
              <div class="d-flex flex-grow-1 flex-column justify-content-center text-center">
                <div>
                  <pf-exchange-job-family-chart></pf-exchange-job-family-chart>
                </div>
              </div>
            </div>
          </div>
          <div class="d-flex flex-grow-2 flex-column justify-content-between">
            <div class="card card-body flex-column justify-content-top flex-grow-1 text-center">
              <h4>Revenue Size</h4>
              <div class="d-flex flex-grow-1 flex-column justify-content-center text-center">
                <div>
                  <pf-exchange-revenue-chart></pf-exchange-revenue-chart>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div class="exchange-dashboard-sidebar" *ngIf="(sidebarVisible$ | async)">
        <pf-chart-detail-component></pf-chart-detail-component>
      </div>
    </div>
  </ng-container>

</pf-page-with-title-bar>

<pf-invite-payfactors-company-modal></pf-invite-payfactors-company-modal>
