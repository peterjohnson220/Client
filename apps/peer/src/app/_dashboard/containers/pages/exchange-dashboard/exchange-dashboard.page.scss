.exchange-dashboard-container {
  &, .exchange-dashboard-chart-container, .exchange-dashboard-job-comparison-grid {
    height: 100%;
  }

  .exchange-dashboard-job-comparison-grid {
    position: relative;
  }
}

.exchange-dashboard-sidebar {
  min-width: 300px;
  max-width: 300px;
}
