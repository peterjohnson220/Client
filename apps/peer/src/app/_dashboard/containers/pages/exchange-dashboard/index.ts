export * from './exchange-dashboard.page';
