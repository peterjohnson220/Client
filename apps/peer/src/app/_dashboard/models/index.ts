export * from './existing-company.model';
