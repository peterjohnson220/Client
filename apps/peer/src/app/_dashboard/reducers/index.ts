import { createSelector, createFeatureSelector } from '@ngrx/store';

// Import root app reducer
import * as fromRoot from 'libs/state/state';

// Import feature reducers
import * as fromExchangeDashboardReducer from './exchange-dashboard.reducer';
import * as fromPeerParticipantsReducer from './peer-participants.reducer';
import * as fromPfCompaniesExchangeRequestReducer from './pf-companies.reducer';
import * as fromAccessExchangeRequestReducer from './access-exchange-request.reducer';
import * as fromExchangeRequestReducer from '../../shared/reducers/exchange-request.reducer';
import * as fromSharedPeerReducer from '../../shared/reducers';

// Feature area state
export interface DashboardState {
  exchangeDashboard: fromExchangeDashboardReducer.State;
  peerParticipants: fromPeerParticipantsReducer.State;
  accessExchangeRequest: fromAccessExchangeRequestReducer.State;
  pfCompaniesExchangeRequest: fromPfCompaniesExchangeRequestReducer.State;
}

// Extend root state with feature area state
export interface State extends fromRoot.State {
  peerDashboard: DashboardState;
}

// Feature area reducers
export const reducers = {
  exchangeDashboard: fromExchangeDashboardReducer.reducer,
  peerParticipants: fromPeerParticipantsReducer.reducer,
  accessExchangeRequest: fromAccessExchangeRequestReducer.reducer,
  pfCompaniesExchangeRequest: fromPfCompaniesExchangeRequestReducer.reducer
};

// Select Feature Area
export const selectFeatureAreaState = createFeatureSelector<DashboardState>('peer_dashboard');

// Feature Selectors
export const selectExchangeDashboardState = createSelector(
  selectFeatureAreaState,
  (state: DashboardState) => state.exchangeDashboard
);

export const selectPeerParticipantsState = createSelector(
  selectFeatureAreaState,
  (state: DashboardState) => state.peerParticipants
);

export const selectAccessExchangeRequestState = createSelector(
  selectFeatureAreaState,
  (state: DashboardState) => state.accessExchangeRequest
);

export const selectPfCompaniesExchangeRequestState = createSelector(
  selectFeatureAreaState,
  (state: DashboardState) => state.pfCompaniesExchangeRequest
);

// Exchange Dashboard Selectors
export const getExchangeDashboardCompanyChartItems = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getCompanyChartItems
);
export const getExchangeDashboardLoadingCompanyChart = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getLoadingCompanyChart
);
export const getExchangeDashboardLoadingCompanyChartError = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getLoadingCompanyChartError
);
export const getExchangeDashboardJobChartItems = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getJobChartItems
);
export const getExchangeDashboardLoadingJobChart = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getLoadingJobChart
);
export const getExchangeDashboardLoadingJobChartError = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getLoadingJobChartError
);
export const getExchangeDashboardIndustryChartItems = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getIndustryChartItems
);
export const getExchangeDashboardLoadingIndustryChart = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getLoadingIndustryChart
);
export const getExchangeDashboardLoadingIndustryChartError = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getLoadingIndustryChartError
);
export const getExchangeDashboardJobFamilyChartItems = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getJobFamilyChartItems
);
export const getExchangeDashboardLoadingJobFamilyChart = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getLoadingJobFamilyChart
);
export const getExchangeDashboardLoadingJobFamilyChartError = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getLoadingJobFamilyChartError
);
export const getExchangeDashboardRevenueChartItems = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getRevenueChartItems
);
export const getExchangeDashboardLoadingRevenueChart = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getLoadingRevenueChart
);
export const getExchangeDashboardLoadingRevenueChartError = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getLoadingRevenueChartError
);
export const getExchangeDashboardDetailChartType = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getDetailChartType
);
export const getExchangeDashboardDetailChartCategory = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getDetailChartCategory
);
export const getExchangeDashboardDetailChartItems = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getDetailChartItems
);
export const getExchangeDashboardLoadingDetailChart = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getLoadingDetailChart
);
export const getExchangeDashboardLoadingDetailChartError = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getLoadingDetailChartError
);
export const getExchangeDashboardSidebarVisible = createSelector(
  selectExchangeDashboardState,
  fromExchangeDashboardReducer.getSidebarVisible
);

// Exchange Request - Access
export const {
  selectAll: getAccessExchangeRequestCandidates
} = fromAccessExchangeRequestReducer.adapter.getSelectors(selectAccessExchangeRequestState);
export const getAccessExchangeRequestCandidatesLoading = createSelector(
  selectAccessExchangeRequestState,
  fromExchangeRequestReducer.getLoading
);
export const getAccessExchangeRequestCandidatesLoadingError = createSelector(
  selectAccessExchangeRequestState,
  fromExchangeRequestReducer.getLoadingError
);
export const getAccessExchangeRequestModalOpen = createSelector(
  selectAccessExchangeRequestState,
  fromExchangeRequestReducer.getModalOpen
);
export const getAccessExchangeRequestRequesting = createSelector(
  selectAccessExchangeRequestState,
  fromExchangeRequestReducer.getRequesting
);
export const getAccessExchangeRequestContext = createSelector(
  selectAccessExchangeRequestState,
  fromAccessExchangeRequestReducer.getLoadingRequestContext
);

// Exchange Request - Access - Peer Participants
export const {
  selectAll: getPeerParticipants
} = fromPeerParticipantsReducer.adapter.getSelectors(selectPeerParticipantsState);

export const getPeerParticipantsLoading = createSelector(
  selectPeerParticipantsState,
  fromPeerParticipantsReducer.getLoading
);

// Exchange Request - pfCompanies
export const {
  selectAll: getPfCompaniesExchangeRequestCandidates
} = fromPfCompaniesExchangeRequestReducer.adapter.getSelectors(selectPfCompaniesExchangeRequestState);
export const getPfCompaniesExchangeRequestCandidatesLoading = createSelector(
  selectPfCompaniesExchangeRequestState,
  fromExchangeRequestReducer.getLoading
);
export const getPfCompaniesExchangeRequestCandidatesLoadingError = createSelector(
  selectPfCompaniesExchangeRequestState,
  fromExchangeRequestReducer.getLoadingError
);
export const getPfCompaniesExchangeRequestModalOpen = createSelector(
  selectPfCompaniesExchangeRequestState,
  fromExchangeRequestReducer.getModalOpen
);
export const getPfCompaniesExchangeRequestRequesting = createSelector(
  selectPfCompaniesExchangeRequestState,
  fromExchangeRequestReducer.getRequesting
);
export const getPfCompaniesExchangeRequestContext = createSelector(
  selectPfCompaniesExchangeRequestState,
  fromSharedPeerReducer.getExchange,
  (exchangeRequestState, exchange) => {
    return {
      Query: exchangeRequestState.searchTerm,
      ExchangeId: exchange ? exchange.ExchangeId : 0
    };
  }
);
