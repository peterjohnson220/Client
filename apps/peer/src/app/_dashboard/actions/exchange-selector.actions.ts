import { Action } from '@ngrx/store';

import { ExchangeListItem } from 'libs/models/peer';

export const LOAD_EXCHANGES  = '[Peer Dashboard/Exchange Selector] Load Exchanges';
export const LOAD_EXCHANGES_SUCCESS  = '[Peer Dashboard/Exchange Selector] Load Exchanges Success';
export const LOAD_EXCHANGES_ERROR = '[Peer Dashboard/Exchange Selector] Load Exchanges Error';

export class LoadExchanges implements Action {
  readonly type = LOAD_EXCHANGES;
}

export class LoadExchangesSuccess implements Action {
  readonly type = LOAD_EXCHANGES_SUCCESS;

  constructor(public payload: ExchangeListItem[]) {}
}

export class LoadExchangesError implements Action {
  readonly type = LOAD_EXCHANGES_ERROR;
}

export type Actions
  = LoadExchanges
  | LoadExchangesSuccess
  | LoadExchangesError;
