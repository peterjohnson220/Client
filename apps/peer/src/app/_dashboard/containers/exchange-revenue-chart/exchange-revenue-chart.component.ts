import { Component, OnInit } from '@angular/core';
import { ActivatedRoute, Router } from '@angular/router';

import { Store } from '@ngrx/store';
import { Observable } from 'rxjs/Observable';

import { Exchange, GetChartRequest, GetDetailChartRequest, ChartItem, ExchangeChartTypeEnum } from 'libs/models';

import * as fromExchangeDashboardActions from '../../actions/exchange-dashboard.actions';
import * as fromPeerDashboardReducer from '../../reducers';
import * as fromSharedPeerReducer from '../../../shared/reducers';

@Component({
  selector: 'pf-exchange-revenue-chart',
  templateUrl: './exchange-revenue-chart.component.html',
  styleUrls: ['./exchange-revenue-chart.component.scss']
})

export class ExchangeRevenueChartComponent implements OnInit {
  exchangeId: number;
  exchange$: Observable<Exchange>;
  revenueChartItems$: Observable<ChartItem[]>;
  loadingRevenueChartItems$: Observable<boolean>;
  loadingRevenueChartItemsError$: Observable<boolean>;

  constructor(
    private route: ActivatedRoute,
    private router: Router,
    private store: Store<fromPeerDashboardReducer.State>,
    private sharedPeerStore: Store<fromSharedPeerReducer.State>
  ) {
    this.exchange$ = this.sharedPeerStore.select(fromSharedPeerReducer.getExchange);
    this.revenueChartItems$ = this.store.select(fromPeerDashboardReducer.getExchangeDashboardRevenueChartItems);
    this.loadingRevenueChartItems$ = this.store.select(fromPeerDashboardReducer.getExchangeDashboardLoadingRevenueChart);
    this.loadingRevenueChartItemsError$ = this.store.select(fromPeerDashboardReducer.getExchangeDashboardLoadingRevenueChartError);
    this.exchangeId = this.route.snapshot.params.id;
  }

  ngOnInit() {
    const getChartRequest: GetChartRequest = {
      ExchangeId: this.exchangeId,
      ChartType: ExchangeChartTypeEnum.Revenue
    };
    this.store.dispatch(new fromExchangeDashboardActions.LoadingRevenueChart(getChartRequest));
  }

  seriesClick(e): void {
    const getDetailChartRequest: GetDetailChartRequest = {
      ExchangeId: this.exchangeId,
      ChartType: ExchangeChartTypeEnum.Revenue,
      Category: e.category
    };
    this.store.dispatch(new fromExchangeDashboardActions.LoadingDetailChart(getDetailChartRequest));
  }
}
