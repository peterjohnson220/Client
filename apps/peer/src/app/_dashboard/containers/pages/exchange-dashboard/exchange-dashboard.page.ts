import { Component, OnInit } from '@angular/core';
import { ActivatedRoute, Router } from '@angular/router';

import { Store } from '@ngrx/store';
import { Observable } from 'rxjs/Observable';

import { Exchange, ExchangeRequestTypeEnum } from 'libs/models';

import * as fromExchangeDashboardActions from '../../../actions/exchange-dashboard.actions';
import * as fromExchangeRequestActions from '../../../../shared/actions/exchange-request.actions';
import * as fromPeerDashboardReducer from '../../../reducers';
import * as fromSharedPeerReducer from '../../../../shared/reducers';

@Component({
  selector: 'pf-exchange-dashboard-page',
  templateUrl: './exchange-dashboard.page.html',
  styleUrls: ['./exchange-dashboard.page.scss']
})
export class ExchangeDashboardPageComponent implements OnInit {
  exchangeId: number;
  exchange$: Observable<Exchange>;
  sidebarVisible$: Observable<boolean>;

  constructor(
    private route: ActivatedRoute,
    private router: Router,
    private store: Store<fromPeerDashboardReducer.State>,
    private sharedPeerStore: Store<fromSharedPeerReducer.State>
  ) {
    this.exchange$ = this.sharedPeerStore.select(fromSharedPeerReducer.getExchange);
    this.sidebarVisible$ = this.store.select(fromPeerDashboardReducer.getExchangeDashboardSidebarVisible);
    this.exchangeId = this.route.snapshot.params.id;
  }

  manageJobsClick(): void {
    this.router.navigate(['manage'], { relativeTo: this.route });
  }

  referCompanyClick(): void {
    this.store.dispatch(new fromExchangeRequestActions.OpenExchangeRequestModal(ExchangeRequestTypeEnum.ReferPayfactorsCompany));
  }

  mapClick(): void {
    this.router.navigate(['map'], { relativeTo: this.route });
  }

  ngOnInit() {
    this.store.dispatch(new fromExchangeDashboardActions.CloseSidebar());
  }
}
