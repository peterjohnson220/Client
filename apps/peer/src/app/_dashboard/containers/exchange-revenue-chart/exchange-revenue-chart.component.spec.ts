import { NO_ERRORS_SCHEMA } from '@angular/core';
import { TestBed, ComponentFixture } from '@angular/core/testing';
import { Router, ActivatedRoute } from '@angular/router';

import { StoreModule, Store, combineReducers } from '@ngrx/store';
import { of } from 'rxjs/observable/of';
import spyOn = jest.spyOn;

import * as fromRootState from 'libs/state/state';
import {  generateMockExchange, ExchangeChartTypeEnum } from 'libs/models';

import { ExchangeRevenueChartComponent } from './exchange-revenue-chart.component';
import * as fromExchangeDashboardActions from '../../actions/exchange-dashboard.actions';
import * as fromPeerDashboardReducer from '../../reducers';

describe('Peer Dashboard - Revenue Chart', () => {
  let fixture: ComponentFixture<ExchangeRevenueChartComponent>;
  let instance: ExchangeRevenueChartComponent;
  let store: Store<fromPeerDashboardReducer.State>;
  let activatedRoute: ActivatedRoute;

  // Configure Testing Module for before each test
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot({
          ...fromRootState.reducers,
          peerDashboard: combineReducers(fromPeerDashboardReducer.reducers)
        }),
      ],
      declarations: [
        ExchangeRevenueChartComponent
      ],
      providers: [
        {
          provide: Router,
          useValue: { navigate: jest.fn() },
        },
        {
          provide: ActivatedRoute,
          useValue: { snapshot: { params: { id : 1 } } },
        },
      ],
      // Shallow Testing
      schemas: [ NO_ERRORS_SCHEMA ]
    });

    store = TestBed.get(Store);
    activatedRoute = TestBed.get(ActivatedRoute);

    fixture = TestBed.createComponent(ExchangeRevenueChartComponent);
    instance = fixture.componentInstance;

    spyOn(store, 'dispatch');
  });

  it('should show the revenue chart and title', () => {
    instance.exchange$ = of(generateMockExchange());

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should dispatch a LoadingRevenueChart action on init', () => {
    instance.exchange$ = of(generateMockExchange());
    const action = new fromExchangeDashboardActions.LoadingRevenueChart({
      ExchangeId: activatedRoute.snapshot.params.id,
      ChartType: ExchangeChartTypeEnum.Revenue
    });

    fixture.detectChanges();

    expect(store.dispatch).toHaveBeenCalledWith(action);
  });


  it('should dispatch a LoadingDetailChart for the revenue on seriesClick', () => {
    fixture.detectChanges();

    const action = new fromExchangeDashboardActions.LoadingDetailChart({
      ExchangeId: activatedRoute.snapshot.params.id,
      ChartType: ExchangeChartTypeEnum.Revenue,
      Category: 'Test'
    });

    instance.seriesClick({ category: action.payload.Category });

    expect(store.dispatch).toHaveBeenCalledWith(action);
  });

});
