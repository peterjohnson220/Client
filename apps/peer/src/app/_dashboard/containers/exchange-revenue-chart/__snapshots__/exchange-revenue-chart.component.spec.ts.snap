// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Peer Dashboard - Revenue Chart should show the revenue chart and title 1`] = `
<pf-exchange-revenue-chart
  exchange$={[Function ScalarObservable]}
  exchangeId={[Function Number]}
  loadingRevenueChartItems$={[Function Store]}
  loadingRevenueChartItemsError$={[Function Store]}
  revenueChartItems$={[Function Store]}
  route={[Function Object]}
  router={[Function Object]}
  store={[Function Store]}
>
  <pf-async-container>
    
  
    
    <kendo-chart>
      
      
      <kendo-chart-legend
        position="bottom"
      />
      
      
      <kendo-chart-area
        background="none"
        height="250"
      />
      
      
      <kendo-chart-tooltip>
        
        
        
        
      
      </kendo-chart-tooltip>
      
      
      <kendo-chart-series>
        
        
        <kendo-chart-series-item
          categoryfield="Category"
          field="Value"
          type="donut"
        >
          
        
        </kendo-chart-series-item>
        
      
      </kendo-chart-series>
      
    
    </kendo-chart>
    
  
  
  </pf-async-container>
  
</pf-exchange-revenue-chart>
`;
